kasa/iot/iotdevice.py

```diff
@@ -436,6 +436,16 @@ def update_from_discover_info(self, info: dict[str, Any]) -> None:
             # by the requires_update decorator
             self._set_sys_info(info)
 
+    @staticmethod
+    def extract_sys_info(info: dict[str, Any]) -> dict[str, Any]:
+        """Return the system info structure."""
+        sysinfo_default = info.get("system", {}).get("get_sysinfo", {})
+        sysinfo_nest = sysinfo_default.get("system", {})
+
+        if len(sysinfo_nest.keys()) > len(sysinfo_default.keys()):
+            return sysinfo_nest
+        return sysinfo_default
```

Comment:
```suggestion
        sysinfo_default = info.get("system", {}).get("get_sysinfo", {})
```
Override the implementation in `IotCamera`